packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -35,6 +35,24 @@ typedef bool RoutePredicate(Route<dynamic> route);
 /// [ModalRoute.removeScopedWillPopCallback], and [WillPopScope].
 typedef Future<bool> WillPopCallback();
 
+/// Signature for a callback which creates a name for each route for the
+/// platform accessibility framework.
+typedef String RouteAccessibilityNameCallback(String routeName, bool isInitialRoute);
+
+/// The default [RouteAccessibilityNameCallback] used in the navigator.
+String _kDefaultAccessibilityNameCallback(String name, bool isInitialRoute) {
+  if (defaultTargetPlatform == TargetPlatform.iOS)
+    return '';
+  if (isInitialRoute)
+    return 'home';
+  if (name.contains('/')) {
+    return name
+      .split('/')
+      .lastWhere((String chunk) => chunk?.isNotEmpty == true, orElse: () => '');
+  }
+  return name;
+}
```

Comment:
If we have this, it must be localized, not hard-coded to English.

We should definitely not expose the actual route names to users, those are internal details.

But I think we shouldn't have this at all. It's up to the developer to provide route names if they want them. We can't come up with a plausible default.